packages/cli/src/utils/astro-config.ts

```diff
@@ -156,9 +167,9 @@ function updateObject(properties: any, object: t.ObjectExpression | undefined):
 
   object ??= t.objectExpression([]);
 
-  for (const property of properties) {
+  for (const property in properties) {
```

Comment:
Do we need to use `Object.hasOwn()` here to make sure that the property is not coming from some prototype? I don't know if that's possible in this case though.

```ts
const bar = {
    baz: 'hello',
};

//////////////////////////

const obj = {
    foo: 'bar',
};

Object.setPrototypeOf(obj, bar);

for (const key in obj) {
    if (Object.hasOwn(obj, key)) {
        console.log(key);
    }
}
```

Without `Object.hasOwn()`, it will print both `foo` and `baz`. Now it only prints `foo`.

Reply:
`properties` is only intended to be a POJO here and is only used with an object literal that we created (it's not a function used with a value produced by some code out of our control).

So that means that the check is not needed.

Happy to add it if we think it could be useful in the future though :+1:

Reply:
No all good if we are sure it's not causing any harm 👍 .